pinot-query-runtime/src/main/java/org/apache/pinot/query/service/QueryDispatcher.java

```diff
@@ -137,25 +137,31 @@ public static List<DataTable> reduceMailboxReceive(MailboxReceiveOperator mailbo
           throw new RuntimeException("Received error query execution result block: "
               + transferableBlock.getDataBlock().getExceptions());
       }
+      if (transferableBlock.isNoOpBlock()) {
+        continue;
+      } else if (transferableBlock.isEndOfStreamBlock()) {
+        return resultDataBlocks;
+      }
```

Comment:
these should be after null check?

Reply:
I don't think so, we shouldn't be adding noops/eos to the results - right?

Reply:
hmm, looking at the code I don't think `TransferableBlock#getDataBlock` can ever return `null` - if the field is null, it either builds the block or throws an exception, otherwise it returns the field.

I'll remove this check altogether since I think it's just misleading. I still think we shouldn't be adding to the result table on metadata blocks

Reply:
:+1: